pilot/pkg/features/pilot.go

```diff
@@ -48,6 +48,13 @@ var (
 		"Limits the number of concurrent pushes allowed. On larger machines this can be increased for faster pushes",
 	).Get()
 
+	// MaxRecvMsgSize The max receive buffer size of gRPC received channel of Pilot in bytes.
+	MaxRecvMsgSize = env.RegisterIntVar(
+		"ISTIO_GPRC_MAXRECVMSGSIZE",
+		4 * 1024 * 1024,
```

Comment:
Do you see an error indicating this? Normally, xds request can not be larger than then default 4MB

Reply:
We introduced EGDS requests in our internal Istio version. The EGDS split each EDS into 1000+ EGDS and suppose you got 500 EDS and 1000 EGDS each, the total EGDS requesting when Proxy starting will be 0.5M. It's quite common for EGDS group name be 10B or more, so the total EGDS request size would easily surpass 4MB.